libc/config/config.json

```diff
@@ -115,6 +115,16 @@
     "LIBC_ADD_NULL_CHECKS": {
       "value": true,
       "doc": "Add nullptr checks in the library's implementations to some functions for which passing nullptr is undefined behavior."
+    },
+    "LIBC_CONF_ALLOC_TYPE": {
+      "value": "LIBC_ALLOC_TYPE_ARENA",
+      "doc": "The implementation used for allocations, acceptable values are LIBC_ALLOC_TYPE_EXTERN, LIBC_ALLOC_TYPE_SCUDO, LIBC_ALLOC_TYPE_ARENA."
```

Comment:
How does this interact with the existing baremetal allocator present in the libc? Immediately it's unusual that it isn't present in a list of allocators.

The allocator doesn't currently work for Linux, but that's something I'm planning to fix in the near future. It's just for lack of a `__morecore` abstraction hooked up to e.g. `sbrk`, plus the plumbing for that sycall.